### PR TITLE
Feature/us8/folder view

### PR DESCRIPTION
**Pull request by dimitriosxmi**
Please take a moment to review this ❤️
Goal: [US8](https://github.com/dimitriosxmi/ArtistsReferenceOrganizer/issues/8) TL; DR: Add a `Folder view` page accessible through a click on an existing folder, creating new entry from a folder sets the folder selection of the entry to the specific folder it was called from.

[Deployment](https://artists-reference-organizer-dxy74hntz-dimitriosxmi.vercel.app/)

```diff
+ Add folder/[folderId].js page
# Displays the entry list of a folder.

! Update Breadcrumb component
# To support the new page and deliver an adaptive breadcrumb.

! Update entries.js api page
# Handles the GET method for specific folderId related entries.

! Update EntryPreviewList component
# Added fetch with folderId router params to get only entries of the specific folder.

! Update FolderPreviewButton component
# Renamed from FolderPreviewIcon to FolderPreviewButton.
# Uses a link now instead of a div.

! Update FolderPreviewList component
# Applies the new FolderPreviewButton component name.

! Update folder.js api page
# Handles the GET method for getting a folder by id.

! Update NewEntryButton component
# Catches from props the folderId and provides it with the new route.

! Update NewEntryForm component
# Catches the folderId router param and provides it to the newly created entry
# if the entry doesn't already have a selected folder Id value.
```

### CODE REVIEW

File: components/EntryPreviewList/index.js [hunk 1]
```diff
@@ -24,6 +24,19 @@ const EntryPreviewList = ({ recentEntriesAmount, hasData }) => {
...
         }
         //#endregion
+        //#region Get entries of specific folder
+        else if (folderId) {
+          const response = await fetch(
+            `/api/entries?selectedFolderId=${folderId}`
+          );
+
+          if (response.ok) {
+            const jsonData = await response.json();
```
Comment (actyvystom) [inline components/EntryPreviewList/index.js:34]: Looks a bit cumbersome - why not using destructuring here?

```js
const { data } =  await response.json();
setEntries(data);
```

**Review by actyvystom — CHANGES_REQUESTED** [review]
Nice work! 🚀 
Please see just a tiny comment 🙂 
- and reviewing your deployment: your `new entry button` has some styling issues, the text is at the lowest z-index (so it gets partially covered by the folder icon)      

Cheers, Thomas

**Comment by dimitriosxmi** [comment]
> Nice work! 🚀 Please see just a tiny comment 🙂
> 
>     * and reviewing your deployment: your `new entry button` has some styling issues, the text is at the lowest z-index (so it gets partially covered by the folder icon)
> 
> 
> Cheers, Thomas

Heyo Thomas,
Thank you for your time and your for your review! 😊 
This is intentional design, I can see perhaps next week if by then I'll be done to consider changing the design of the entire app or parts like this  `NewEntryButton` components 😋

File: components/EntryPreviewList/index.js (hunk 1 above)
Comment (dimitriosxmi) [reply components/EntryPreviewList/index.js:34]: Oh duh I'm dumb, thanks for the input!
I think there's somewhere also some router parameters I didn't destructure I'll tend to it right away! 😄

**Comment by MichanNan** [comment]
LGTM i really like the comments you write, that makes it easier to read the code!

**Pull request merged**